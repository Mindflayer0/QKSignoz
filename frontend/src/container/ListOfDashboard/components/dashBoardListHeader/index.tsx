import { PlusOutlined } from '@ant-design/icons';
import { Dropdown, MenuProps, Row, Typography } from 'antd';
import TextToolTip from 'components/TextToolTip';
import { NewDashboardState } from 'container/ListOfDashboard';
import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';

import { ButtonContainer, NewDashboardButton } from '../../styles';
import { getDashBoardText, getMenuItems } from './helpers';

function DashBoardListHeader({
	isDashboardListLoading,
	newDashboard,
	newDashboardState,
	createNewDashboard,
	onModalHandler,
	onNewDashboardHandler,
}: Props): JSX.Element {
	const { t } = useTranslation('dashboard');

	const menu: MenuProps = useMemo(
		() => ({
			items: getMenuItems({
				createNewDashboard,
				isDashboardListLoading,
				onModalHandler,
				onNewDashboardHandler,
				t,
			}),
		}),
		[
			createNewDashboard,
			isDashboardListLoading,
			onModalHandler,
			onNewDashboardHandler,
			t,
		],
	);

	return (
		<Row justify="space-between">
			<Typography>Dashboard List</Typography>

			<ButtonContainer>
				<TextToolTip
					text="More details on how to create dashboards"
					url="https:signoz.io/docs/userguide/dashboards"
				/>
				{newDashboard && (
					<Dropdown
						disabled={isDashboardListLoading}
						trigger={['click']}
						menu={menu}
					>
						<NewDashboardButton
							icon={<PlusOutlined />}
							type="primary"
							loading={newDashboardState.loading}
							danger={newDashboardState.error}
						>
							{getDashBoardText(newDashboardState)}
						</NewDashboardButton>
					</Dropdown>
				)}
			</ButtonContainer>
		</Row>
	);
}

interface Props {
	newDashboard: boolean;
	isDashboardListLoading: boolean;
	newDashboardState: NewDashboardState;
	createNewDashboard: boolean;
	onNewDashboardHandler: () => Promise<void>;
	onModalHandler: (uploadedGrafana: boolean) => void;
}

export default DashBoardListHeader;
