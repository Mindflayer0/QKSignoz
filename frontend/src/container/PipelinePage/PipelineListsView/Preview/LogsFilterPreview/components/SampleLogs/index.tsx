import './styles.scss';

import { Time } from 'container/TopNav/DateTimeSelection/config';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';

import LogsList from '../../../components/LogsList';
import useSampleLogs from '../../../hooks/useSampleLogs';

function SampleLogs({ filter, timeInterval }: SampleLogsProps): JSX.Element {
	const { isLoading, isError, logs } = useSampleLogs({
		filter,
		timeInterval,
		count: 5,
	});

	if (isError) {
		return (
			<div className="sample-logs-notice-container">
				could not fetch logs for filter
			</div>
		);
	}

	if (isLoading) {
		return <div className="sample-logs-notice-container">Loading...</div>;
	}

	if ((filter?.items?.length || 0) < 1) {
		return (
			<div className="sample-logs-notice-container">Please select a filter</div>
		);
	}

	if (logs.length < 1) {
		return <div className="sample-logs-notice-container">No logs found</div>;
	}

	return <LogsList logs={logs} />;
}

interface SampleLogsProps {
	filter: TagFilter;
	timeInterval: Time;
}

export default SampleLogs;
