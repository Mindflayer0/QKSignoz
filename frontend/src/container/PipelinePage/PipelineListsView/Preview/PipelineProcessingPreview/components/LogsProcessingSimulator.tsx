import { Button } from 'antd';
import { useState } from 'react';
import { ILog } from 'types/api/logs/log';
import { PipelineData } from 'types/api/pipeline/def';

function LogsProcessingSimulator({
	inputLogs,
	pipeline,
}: LogsProcessingSimulatorProps): JSX.Element {
	const [showResult, setShowResult] = useState<boolean>(false);
	const simulate = (): void => setShowResult(true);

	if (!showResult) {
		return (
			<div>
				<Button type="primary" onClick={simulate}>
					Simulate Processing
				</Button>
			</div>
		);
	}

	return <div>{JSON.stringify({ inputLogs, pipeline })}</div>;
}

export interface LogsProcessingSimulatorProps {
	inputLogs: ILog[];
	pipeline: PipelineData;
}

export default LogsProcessingSimulator;
