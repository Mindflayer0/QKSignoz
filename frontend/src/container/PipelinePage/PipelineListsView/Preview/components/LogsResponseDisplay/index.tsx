import { LogsResponse } from '../../types';
import LogsList from '../LogsList';

function LogsResponseDisplay({
	response,
}: LogsResponseDisplayProps): JSX.Element {
	const { isLoading, isError, logs } = response;

	if (isError) {
		return (
			<div className="sample-logs-notice-container">There was an error.</div>
		);
	}

	if (isLoading) {
		return <div className="sample-logs-notice-container">Loading...</div>;
	}

	if (logs.length < 1) {
		return <div className="sample-logs-notice-container">No logs found</div>;
	}

	return <LogsList logs={logs} />;
}

export interface LogsResponseDisplayProps {
	response: LogsResponse;
}

export default LogsResponseDisplay;
