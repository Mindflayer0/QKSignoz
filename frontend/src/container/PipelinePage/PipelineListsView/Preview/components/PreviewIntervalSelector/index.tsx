import './styles.scss';

import { Select } from 'antd';
import {
	initialFilters,
	initialQueriesMap,
	PANEL_TYPES,
} from 'constants/queryBuilder';
import {
	RelativeDurationOptions,
	Time,
} from 'container/TopNav/DateTimeSelection/config';
import { useGetQueryRange } from 'hooks/queryBuilder/useGetQueryRange';
import { cloneDeep } from 'lodash-es';
import { useMemo } from 'react';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';
import { LogsAggregatorOperator } from 'types/common/queryBuilder';

function MatchedLogsCount({
	filter,
	timeInterval,
}: MatchedLogsCountProps): JSX.Element {
	const query = useMemo(() => {
		const q = cloneDeep(initialQueriesMap.logs);
		q.builder.queryData[0] = {
			...q.builder.queryData[0],
			filters: filter || initialFilters,
			aggregateOperator: LogsAggregatorOperator.COUNT,
		};
		return q;
	}, [filter]);

	const result = useGetQueryRange({
		graphType: PANEL_TYPES.TABLE,
		query,
		selectedTime: 'GLOBAL_TIME',
		globalSelectedInterval: timeInterval,
	});

	if (result.isFetched) {
		const count =
			result?.data?.payload?.data?.newResult?.data?.result?.[0]?.series?.[0]
				?.values?.[0]?.value;
		return (
			<div className="logs-filter-preview-matched-logs-count">
				{count} matches in
			</div>
		);
	}
	return <div />;
}

interface MatchedLogsCountProps {
	filter: TagFilter;
	timeInterval: Time;
}

function PreviewIntervalSelector({
	previewFilter,
	value,
	onChange,
}: PreviewIntervalSelectorProps): JSX.Element {
	const isEmptyFilter = (previewFilter?.items?.length || 0) < 1;

	return (
		<div className="logs-filter-preview-time-interval-summary">
			{!isEmptyFilter && (
				<MatchedLogsCount filter={previewFilter} timeInterval={value} />
			)}
			<div>
				<Select
					onSelect={(value: unknown): void => onChange(value as Time)}
					value={value}
				>
					{RelativeDurationOptions.map(({ value, label }) => (
						<Select.Option key={value + label} value={value}>
							{label}
						</Select.Option>
					))}
				</Select>
			</div>
		</div>
	);
}

interface PreviewIntervalSelectorProps {
	value: Time;
	onChange: (interval: Time) => void;
	previewFilter: TagFilter;
}

export default PreviewIntervalSelector;
