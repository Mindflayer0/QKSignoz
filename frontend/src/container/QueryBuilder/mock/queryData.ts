import { initialQueryBuilderFormValuesMap } from 'constants/queryBuilder';
import { IBuilderQuery } from 'types/api/queryBuilder/queryBuilderData';
import { DataSource } from 'types/common/queryBuilder';

export const queryMockData: IBuilderQuery = {
	dataSource: DataSource.METRICS,
	queryName: 'A',
	aggregateOperator: 'noop',
	aggregateAttribute: {
		key: 'signoz_latency_count',
		dataType: 'float64',
		type: 'tag',
		isColumn: true,
	},
	filters: {
		items: [
			{
				id: '8fac746b',
				key: {
					key: 'resource_signoz_collector_id',
					dataType: 'int64',
					isColumn: false,
					type: 'tag',
				},
				op: '=',
				value: ['1a5d3cc2-4b3e-4c7c-ad07-c4cdd739d1b9'],
			},
			{
				id: '8fadtr46b',
				key: {
					key: 'service_name',
					dataType: 'string',
					isColumn: false,
					type: 'tag',
				},
				op: 'IN',
				value: ['frontend', 'operation', 'ser'],
			},
		],
		op: 'AND',
	},
	expression: '',
	disabled: false,
	having: [],
	stepInterval: 30,
	limit: 10,
	orderBy: [],
	groupBy: [],
	legend: '',
	reduceTo: 'sum',
};

export const formulaMockData = {
	queryName: 'C',
	label: 'Formula',
	expression: 'sum(A, B)',
	legend: 'Total',
	disabled: false,
};

export const valueWithAttributeAndOperator: IBuilderQuery = {
	...initialQueryBuilderFormValuesMap.logs,
	dataSource: DataSource.LOGS,
	aggregateOperator: 'SUM',
	aggregateAttribute: {
		isColumn: false,
		key: 'bytes',
		type: 'tag',
		dataType: 'float64',
	},
};

export const querySearchResult = ['frontend', 'operation', 'ser'];
