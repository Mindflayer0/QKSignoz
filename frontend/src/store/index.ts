import { applyMiddleware, compose, createStore } from 'redux';
import type { ThunkMiddleware } from 'redux-thunk';
import thunk from 'redux-thunk';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	// @TODO Add Type for AppActions also
	composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<AppState>)),
);

export default store;
