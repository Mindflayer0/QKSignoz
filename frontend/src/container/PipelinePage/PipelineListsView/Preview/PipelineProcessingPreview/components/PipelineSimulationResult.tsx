import { ILog } from 'types/api/logs/log';
import { PipelineData } from 'types/api/pipeline/def';

import LogsResponseDisplay from '../../components/LogsResponseDisplay';
import usePipelinePreview from '../../hooks/usePipelinePreview';

function PipelineSimulationResult({
	inputLogs,
	pipeline,
}: PipelineSimulationResultProps): JSX.Element {
	const simulationResult = usePipelinePreview({ pipeline, logs: inputLogs });

	return <LogsResponseDisplay response={simulationResult} />;
}

export interface PipelineSimulationResultProps {
	inputLogs: ILog[];
	pipeline: PipelineData;
}

export default PipelineSimulationResult;
