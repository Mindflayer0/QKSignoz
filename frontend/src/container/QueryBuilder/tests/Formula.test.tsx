import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { Formula } from '../components';
import { formulaMockData } from '../mock/queryData';

describe('QueryBuilder: Formula Component', () => {
	it('it should render Formula', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<Formula formula={formulaMockData} index={0} />
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
