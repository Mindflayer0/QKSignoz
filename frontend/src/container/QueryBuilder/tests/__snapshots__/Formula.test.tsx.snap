// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilder: Formula Component it should render Formula 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: 0.5rem;
  right: 0.9375rem;
  z-index: 1;
  cursor: pointer;
  opacity: 0.45;
  width: 1.3125rem;
  height: 1.3125rem;
}

.c1 svg {
  width: 100%;
  height: 100%;
}

.c0 {
  padding-right: 3rem;
}

.c2 {
  min-width: 2rem;
  height: 2.25rem;
  padding: 0.43rem;
  border-radius: 0.375rem;
  pointer-events: default;
}

<div
    class="ant-row c0 css-dev-only-do-not-override-1i536d8"
    style="margin-top: -7.5px; margin-bottom: -7.5px;"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle c1"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
        />
        <path
          d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
      </svg>
    </span>
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-1i536d8"
      style="padding-top: 7.5px; padding-bottom: 7.5px;"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-primary c2"
        type="button"
      >
        <span
          aria-label="eye"
          class="anticon anticon-eye"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="eye"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
            />
          </svg>
        </span>
        <span>
          C
        </span>
      </button>
    </div>
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-1i536d8"
      style="padding-top: 7.5px; padding-bottom: 7.5px;"
    >
      <textarea
        class="ant-input css-dev-only-do-not-override-1i536d8"
        name="expression"
        rows="2"
      >
        sum(A, B)
      </textarea>
    </div>
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-1i536d8"
      style="padding-top: 7.5px; padding-bottom: 7.5px;"
    >
      <span
        class="ant-input-group-wrapper css-dev-only-do-not-override-1i536d8"
      >
        <span
          class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1i536d8"
        >
          <span
            class="ant-input-group-addon"
          >
            Legend Format
          </span>
          <input
            class="ant-input css-dev-only-do-not-override-1i536d8"
            name="legend"
            type="text"
            value="Total"
          />
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
